IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDKeychainUtil.h"
+#import "MSIDWorkPlaceJoinUtil.h"
+#import "MSIDWorkPlaceJoinUtilBase.h"
+#import "MSIDWorkPlaceJoinConstants.h"
+#import "MSIDAssymetricKeyPairWithCert.h"
+
+NSString *const MSID_DEVICE_INFORMATION_UPN_ID_KEY        = @"userPrincipleName";
+NSString *const MSID_DEVICE_INFORMATION_AAD_DEVICE_ID_KEY = @"aadDeviceIdentifier";
+NSString *const MSID_DEVICE_INFORMATION_AAD_TENANT_ID_KEY = @"aadTenantIdentifier";
+
+@implementation MSIDWorkPlaceJoinUtilBase
+
++ (NSString *_Nullable)getWPJStringDataForIdentifier:(nonnull NSString *)identifier
+                                             context:(id<MSIDRequestContext>_Nullable)context
+                                               error:(NSError*__nullable*__nullable)error
+{
+    #if TARGET_OS_IPHONE
```

Comment:
this part can be passed in by the subclass, right? That way you wouldn't need an "ifdef"
You can have this API accept additional platform query parameters, which would provide access group on iOS and empty dictionary for now on macOS (but in the future it might have something else for macOS as well)

Reply:
In latest commit, I have moved all code in MSIDWorkPlaceJoinUtilBase and now getWPJStringDataForIdentifier getting called from getRegisteredDeviceMetadataInformation in same file.
Now subclass don't have any code around this PR and MSIDWorkPlaceJoinUtil is now unchanged.
so IMO this portion of code can be conditional.

Reply:
It will be easier for readability to have it in specific platform files. Conditionals tend to grow large over time and makes it less readable. Since we already have separate iOS and macOS files, it would make more sense there to have platform specific attributes.